### ISSUE TITLE
eliminate all format errors

### ISSUE THREAD

**Issue by jeffhammond** [ParRes/Kernels#39]
This issues will be closed when the PRK can be built with `CFLAGS=-Wformat -Werror`.  We have an enormous number of cases where we pass `long` or a `long long` to `printf(%d)`.

**Comment by rfvander** [comment]
These errors arose recently when we turned many integers into longs to avoid integer overflow triggered by large problem sizes. It has always been my goal to compile without ANY warnings, and for a long time that was indeed the case, except for Charm++ builds, since header files created by Charm++ are not under the control of the programmer. We now need to revisit all these changes.

**Comment by jeffhammond** [comment]
@rfvander Yeah, I know that.  I just add issues every time I think of something we need to do, since it gives me a place to look when I need something to code :-)

**Comment by SamTWhite** [comment]
@rfvander Some code-gen formatting issues were fixed recently in Charm++, and I don't see any issues building the PRK with "-Wformat -Werror" using the current master branch of Charm++ built on icc or gcc.

**Comment by rfvander** [comment]
That is excellent. Now we should still learn to build in a less verbose mode--same for Grappa--so that by default we don't get all this runtime information.

**Comment by SamTWhite** [comment]
By runtime information I assume you mean this?

Charmrun> scalable start enabled. 
Charmrun> started all node programs in 0.595 seconds.
Charm++> Running in non-SMP mode: numPes 4
Converse/Charm++ Commit ID: v6.7.0-30-g80080fa
Charm++> scheduler running in netpoll mode.
CharmLB> Load balancer assumes all CPUs are same.
Charm++> Running on 1 unique compute nodes (8-way SMP).
Charm++> cpu topology info is gathered in 0.001 seconds.

We could add such a flag, but there is none right now.

**Comment by rfvander** [comment]
Indeed. This is useful information, but clutters the output of batch runs.

**Comment by SamTWhite** [comment]
Our 6.7.1 release contains a '++quiet' runtime option to silence this output.
